### ISSUE TITLE
Lighters and matchboxes are unable to light items/tiles ablaze

### ISSUE THREAD

**Issue by Irets** [CleverRaven/Cataclysm-DDA#65735]
### Describe the bug

On the latest developement version of cdda, matchboxes and lighters are unable to light items ablaze by [a]ctivating them and choosing an adjacent tile.

### Attach save file

[Helvete-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/11515705/Helvete-trimmed.tar.gz)
Here is a save of mine where on your right is a pile of butchery refuse, leaves and a zombie child corpse.
I currently have a matchbox & lighter with me which are unable to light the fire, as it says "There's nothing to light there".

A quick way of testing this out on your system is to create a new world, choose Punk Rock Dude as your character as he spawns with a lighter and then attempt to light something flammable on fire through the [a]ctivation menu.

### Steps to reproduce

1. Spawn into a world with either matchbox or a lighter
2. [a]ctivate said item and attempt to light something adjacent to you ablaze
3. Fail the above and be met with "There's nothing to light there".

![maim_1684489690](https://github.com/CleverRaven/Cataclysm-DDA/assets/50279814/65b6c0ee-c5ae-4119-b08e-e7208d889c44)

### Expected behavior

I except items or the adjacent tile to light on fire.

### Screenshots

![maim_1684489690](https://github.com/CleverRaven/Cataclysm-DDA/assets/50279814/43826558-f2a7-40f4-81a7-3330fa0491b6)
![maim_1684490329](https://github.com/CleverRaven/Cataclysm-DDA/assets/50279814/ea037bd8-e285-4c26-96b5-99c038b97b1c)

### Versions and configuration

OS : Gentoo Linux
Game version : 0.G-1998-gc3ffd0cbf0-dirty
Graphics : either
Ingame language : English
Mods loaded : dda, no_npc_food, personal_portal_storms, no_fungal_growth, package_bionic_professions

### Additional context

I am very certain that in a previous version of the game, I was able to burn houses and light piles of butchery refuse on fire, but not in this latest developement version.

**Comment by NetSysFire** [comment]
I wonder if this could be related to #65536.

**Comment by Irets** [comment]
Through additional testing, it became apparent that the issue doesn't affect trees.
Those can be lit with either matches or a lighter, while most other items or tiles can not.

**Comment by RenechCDDA** [comment]
Leaves are `Vegetable Matter` and should probably be `Dry Plant`. They should also get the TINDER flag. I can take a quick shake at this soon.

**Comment by Irets** [comment]
Thank you, @RenechCDDA.
Should butchery refuse and/or zombie corpses (which are clothed) be flammable?

I very clearly remember lighting dissected corpses with pieces of various clothing ablaze post-dissection, but this was in a older version of the game.

**Comment by RenechCDDA** [comment]
The fleshy bits are not flammable by themselves and shouldn't be. They *do* contribute to more intense fires. (Indeed in the older versions corpses and flesh would go up like a box of matches)

Carpet seems to be FLAMMABLE_HARD, which is why you can't light it directly. It's kind of weird it's less flammable than like, walls, but all the terrain interactions have not been touched in quite a long time. (if you deconstructed it into an item, it'd burn quite well)

**Comment by RenechCDDA** [comment]
No, not really. Cotton clothing is the most flammable of the materials present and it's not going to go up if exposed to a match. It needs some burn to get going.

Not saying there *aren't* scenarios where a pile of clothing couldn't catch fire from a single match IRL, but it's pretty hard to do. There generally needs to be some tinder or accelerant.

**Comment by fairyarmadillo** [comment]
Fresh leaves probably shouldn't be flammable. We should have a green and a dead leaf variant, so there's more tinder in the fall.

I did notice that newspaper pages weren't able to be lit a couple of days ago. Probably anything made of paper should go right up.

**Comment by LordKaaven** [comment]
![fire](https://github.com/CleverRaven/Cataclysm-DDA/assets/66031499/856e9c66-7fb3-418c-90c9-17df32ff25e4)
Paper items, such as this map, also do not seem to be flammable anymore. If we cannot start fire with paper, then what?

**Comment by Irets** [comment]
There seems to be confusion, at least on my part.
It is true that certain items are not flammable anymore and it should be addressed, but what I think threw me off the most, was the fact that I was used to being able to burn pretty much anything in older game versions.

The error text being "There's nothing to light there." is really confusing.
Of course there is *something* to light there, but it may not be flammable enough.

I propose that we could have a check when attempting to light something on fire.
ie. when using a lighter or matches on a tile, it would report something along the lines "you aren't able to get a fire going" to hint at the items/tile not being flammable enough. If the tile doesn't have any items and consist of non-flammable material, then print the original "There's nothing to light there".

The logic for this seems to be defined in activity_handlers.cpp, line 1718.

What do you think?

**Comment by NetSysFire** [comment]
I discussed this on IRC with irets and we both agree that it mostly looks like a misleading error message here. Perhaps something like "None of the items there can be lit on its own.  Try adding some wood."
That message is something I made up on the spot and I am not perfectly content with the latter part.

https://github.com/CleverRaven/Cataclysm-DDA/blob/470485155c458d28001ef0b038fe26977bd00995/src/activity_handlers.cpp#L1710-L1722

However, being unable to light paper is definitely something that looks unintended. Spawn e.g some newspaper, try to light it, get the message that there is nothing to light.

@RenechCDDA, could you perhaps look into the paper issue?

**Comment by RenechCDDA** [comment]
It definitely looks like I put the wrong values in for paper.

**Comment by RenechCDDA** [comment]
I agree with updating the string in C++, but I also don't know what it should say. Maybe something like `There's nothing there that burns well.`? Ultimately it needs to convey that the materials' `fuel` values are <1.0. Of course there's also TINDER items which should be lightable regardless and.... well, my preferred fix is a bit bigger than just changing one message.

I haven't included any C++ changes in the linked PR, I'll mull it over for a bit but no guarantees I resolve it. (And hopefully if I do decide on changes life won't kidnap me for a few weeks again)
